### ISSUE TITLE
bpftrace -l lists duplicate probles

### ISSUE THREAD

**Issue by andreasgerstmayr** [iovisor/bpftrace#1581]
### What reproduces the bug?
```
$ sudo ./src/bpftrace -l | sort | uniq -c -d | sort -n -r
    147 kprobe:__ia32_sys_old_shmctl
    119 kprobe:__ia32_compat_sys_io_pgetevents_time32
     42 kprobe:__ia32_sys_ipc
     32 kprobe:__ia32_sys_uselib
     21 kprobe:__ia32_sys_subpage_prot
     18 kprobe:type_show
     17 kprobe:name_show
     17 kprobe:modalias_show
     15 kprobe:sys_ni_syscall
     13 kprobe:__ia32_sys_old_semctl
     13 kprobe:__ia32_sys_old_msgctl
     12 kprobe:__ia32_sys_ppoll_time32
     11 kprobe:init_once
      9 kprobe:print_cpus_offline
      8 kprobe:state_show
      7 kprobe:event_show
      6 kprobe:umask_show
      6 kprobe:new_id_store
      6 kprobe:__ia32_sys_pciconfig_iobase
      5 kprobe:version_show
[...]
```

In total 1407 probes are duplicated (happens with `kprobe` and `kfunc`). Is this expected?

I looked in the code, and it turns out the kprobe duplicates are coming from `/sys/kernel/debug/tracing/available_filter_functions`. Not sure if they should have been de-duplicated there already, or should be de-duplicated in bpftrace?

### bpftrace --info
```
System
  OS: Linux 5.8.15-201.fc32.x86_64 #1 SMP Thu Oct 15 15:56:44 UTC 2020
  Arch: x86_64

Build
  version: v0.11.0-212-g4808
  LLVM: 10.0.1
  foreach_sym: yes
  unsafe uprobe: no
  bfd: yes
  bpf_attach_kfunc: yes
  bcc_usdt_addsem: yes
  libbpf: yes
  libbpf btf dump: yes
  libbpf btf dump type decl: yes

Kernel helpers
  probe_read: yes
  probe_read_str: yes
  probe_read_user: yes
  probe_read_user_str: yes
  probe_read_kernel: yes
  probe_read_kernel_str: yes
  get_current_cgroup_id: yes
  send_signal: yes
  override_return: no
  get_boot_ns: yes
  dpath: no

Kernel features
  Instruction limit: 1000000
  Loop support: yes
  btf (depends on Build:libbpf): yes
  map batch (depends on Build:libbpf): yes

Map types
  hash: yes
  percpu hash: yes
  array: yes
  percpu array: yes
  stack_trace: yes
  perf_event_array: yes

Probe types
  kprobe: yes
  tracepoint: yes
  perf_event: yes
  kfunc: yes
```

**Comment by mmisono** [comment]
Apparently available_filter_functions contain duplicate entries.

```
% sudo cat /sys/kernel/debug/tracing/available_filter_functions | sort | uniq -c -d | sort -n -r | head
    131 __x32_compat_sys_getsockopt
    129 __x32_compat_sys_io_pgetevents_time32
     57 __ia32_sys_subpage_prot
     43 __ia32_sys_ipc
     31 __ia32_sys_old_shmctl
     17 sys_ni_syscall
     17 __ia32_sys_old_msgctl
     16 type_show
     15 __ia32_sys_old_semctl
     14 name_show
[...]
```

**Comment by andreasgerstmayr** [comment]
Yes, the question is now: should they have been de-duplicated in the kernel already, or should they be de-duplicated in bpftrace? Some `kfunc`s are also duplicated, now sure yet if the reason is similar or there is another cause for this.

**Comment by viktormalik** [comment]
kfuncs are duplicated if BTF contains duplicate function entries (there is indeed a small number of such functions).

Anyway, this problem is fixed by #1549 as it de-duplicates listed entries.
